### ISSUE TITLE
Raven::Instance still uses global context and configuration

### ISSUE THREAD

**Issue by singpolyma-shopify** [getsentry/raven-ruby#597]
When creating a `Raven::Event`, it just calls `from_message` or `from_exception` and those use `Raven.context` and `Raven.configuration` (there appear to be some ways to override this behaviour, but `Raven::Instance` is not using them)

**Comment by nateberkopec** [comment]
Yeah, this is related to #580. We don't have great testing around isolating Raven::Instance right now, it's something I want to improve for the next release. Thanks for pointing it out.
